### PR TITLE
*: fix broken links

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

Fix broken links:

- The link to the node-mysql2 documentation has been updated. The previous URL `https://github.com/sidorares/node-mysql2/tree/master/documentation/en` was removed in https://github.com/sidorares/node-mysql2/pull/2326. It now points to the current documentation site: https://sidorares.github.io/node-mysql2/docs/documentation.
- The link `https://docs.pingcap.com/tidb/stable/dev-guide-sample-application-java` is leading to a 404 error because `develop/dev-guide-sample-application-java.md` was removed without any redirect aliases being added.
- The TiDB Lightning package `https://github.com/pingcap/tidb/tree/master/br/pkg/lightning` was modified to `https://github.com/pingcap/tidb/tree/master/lightning` in https://github.com/pingcap/tidb/pull/52247.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-faq.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ DM 仅支持解析标准版本的 MySQL/MariaDB 的 binlog，对于阿里云 RDS
 - 阿里云 RDS
     - 即使上游表没有主键，阿里云 RDS 的 binlog 中也会包含隐藏的主键列，与上游表结构不一致。
 - 华为云 RDS
-    - 不支持，详见：[华为云数据库 RDS 是否支持直接读取 Binlog 备份文件](https://support.huaweicloud.com/rds_faq/rds_faq_0210.html)。
+    - 不支持。
```
Comment (Oreoxmt) [inline dm/dm-faq.md:17]:
```suggestion
    - 不支持。详见：[华为云数据库 RDS 是否支持直接读取 Binlog 备份文件](https://support.huaweicloud.com/en-us/rds_faq/rds_faq_0210.html)。
```

File: dm/dm-faq.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ DM 仅支持解析标准版本的 MySQL/MariaDB 的 binlog，对于阿里云 RDS
 - 阿里云 RDS
     - 即使上游表没有主键，阿里云 RDS 的 binlog 中也会包含隐藏的主键列，与上游表结构不一致。
 - 华为云 RDS
-    - 不支持，详见：[华为云数据库 RDS 是否支持直接读取 Binlog 备份文件](https://support.huaweicloud.com/rds_faq/rds_faq_0210.html)。
+    - 不支持。详见：[华为云数据库 RDS 是否支持直接读取 Binlog 备份文件](https://support.huaweicloud.com/en-us/rds_faq/rds_faq_0210.html)。
```
Comment (Oreoxmt) [inline dm/dm-faq.md:17]:
```suggestion
    - 不支持，详见：[华为云数据库 RDS 是否支持直接读取 Binlog 备份文件](https://support.huaweicloud.com/en-us/rds_faq/rds_faq_0210.html)。
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 3]
```diff
@@ -44,7 +44,7 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
 
     <div label="macOS">
 
-    对于 macOS 操作系统，如果你没有安装 Homebrew，请参考 [Homebrew 官网](https://brew.sh/index_zh-cn)进行安装。
+    对于 macOS 操作系统，如果你没有安装 Homebrew，请参考 [Homebrew 官网](https://brew.sh/index)进行安装。
```
Comment (lilin90) [inline develop/dev-guide-build-cluster-in-cloud.md:47]:
```suggestion
    对于 macOS 操作系统，如果你没有安装 Homebrew，请参考 [Homebrew 官网](https://brew.sh/zh-cn/)进行安装。
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**